tools/lisa-build-asset

```diff
@@ -57,25 +73,147 @@ def list_assets(recipe_dir):
-            recipe=recipe,
-            action=action,
+    # List the required Alpine packages.
+    try:
+        alpine_dependencies = get_asset_property(
+            asset, arch, "ALPINE_BUILD_DEPENDENCIES",
+            build_dir=build_dir,
+            recipe_dir=recipe_dir,
+            arch_chroot_dir=arch_chroot_dir
         )
-        cmd = ['bash', '-c', script]
-        subprocess.check_call(cmd, env=env, cwd=build_dir)
+    except ValueError:
+        alpine_dependencies = []
+    else:
+        alpine_dependencies = ('-p', ' '.join(map(quote, alpine_dependencies)))
+
+    # List the required env variables.
+    env_args = ('-k', ' '.join(env.keys())) if env else []
+
+    # Install the chroot.
+    subprocess.check_call(
+        [
+            'sudo', os.path.join(get_alpine_dir(), 'alpine-chroot-install'),
+            '-d', arch_chroot_dir,
+            '-i', LISA_HOME,
+            '-a', QEMU_ARCH_NAMES.get(arch, arch),
+            *alpine_dependencies,
+            *env_args,
+        ],
+        cwd=build_dir,
+    )
 
-ARCHITECTURES = ['arm64', 'armeabi', 'x86_64']
+    def bind_mount(path):
+        # join() will ignore everything before an absolute path so make it
+        # relative first
+        rel = os.path.relpath(path, start='/')
+        path_ = os.path.join(arch_chroot_dir, rel)
+        subprocess.check_call(['sudo', 'mkdir', '-p', path_])
+        subprocess.check_call(['sudo', 'mount', '--bind', path, path_])
+
+    # If build_dir is not in LISA_HOME, bind it aswell.
+    if not build_dir.startswith(LISA_HOME):
+        bind_mount(build_dir)
+
+    try:
+        yield arch_chroot_dir
+    finally:
+        # Clean-up the chroot
+        subprocess.check_call([os.path.join(arch_chroot_dir, 'destroy'), '--remove'])
+
+def make(asset, arch, actions, build_dir, recipe_dir, toolchain=None, arch_chroot_dir=None, get_output=False):
+    if arch_chroot_dir:
+        # Since the build is native, no need for a toolchain.
+        toolchain = None
+
+    if get_output:
+        # make() is sometimes expected to return the output of the 'actions'.
+        # If so, use check_output() to exectute the 'actions'.
+        output = dict()
+        def run_cmd(action, *args, **kwargs):
+            output[action] = subprocess.check_output(*args, **kwargs).decode('utf-8')
+    else:
+        output = None
+        def run_cmd(action, *args, **kwargs):
+            subprocess.check_call(*args, **kwargs)
+
+    recipe = get_asset_recipe(asset, recipe_dir)
+    env = get_env(
+        asset=asset,
+        arch=arch,
+        build_dir=build_dir,
+        recipe=recipe,
+        toolchain=toolchain
+    )
+
+    for action in actions:
+        print(f'\nMaking {arch} {asset} {action} in: {build_dir}\n')
+
+        cmd = ['bash', '-c', f'source {recipe} && {action}']
+
+        if arch_chroot_dir and action in ('build', 'install'):
+            # Notify the recipe the build is on alpine.
+            cmd[-1] = f'{cmd[-1]} alpine'
```

Comment:
Can be replaced with an environment variable to indicate we are building on alpine